feat(HTTP): Refresh request method pages (#35745)
mdn/content: files/en-us/web/http/methods/delete/index.md

{{HTTPSidebar}}

> The **HTTP `DELETE` request method** asks the server to delete a specified resource.

Reviewer: ```suggestion
The **`DELETE`** HTTP method asks the server to delete a specified resource.
```

> Content sent in a `DELETE` request has no defined semantics, and therefore should be omitted.

Reviewer: I'm OK with "Content" here. But given our previous discussions on the topic should we use "Message body" here?

```suggestion
The `DELETE` method has no defined semantics for the message body, so this should be empty.
```

## Syntax

```http
DELETE <request-target>["?"<query>] HTTP/1.1
```

- `<request-target>`
  - : Identifies the target resource of the request when combined with the information provided in the {{HTTPHeader("Host")}} header.
    This is an absolute path (e.g., `/path/to/file.html`) in requests to an origin server, and an absolute URL in requests to proxies (e.g., `http://www.example.com/path/to/file.html`).

> - `<query>`

Reviewer: You could use `{{optional_inline}} for optional things. The BNF and your text means this isn't "needed" but on the other hand it's a common visual cue on the docs. Up to you - I have no preference.

Author: Good idea, tnx

> DELETE /file.html HTTP/1.1

Reviewer: Not sure what a syntax box should look like, but I would have perhaps expected:

```suggestion
DELETE <url> HTTP/1.1
```

Or `<url-of-resource>`

Reviewer: Then an explanation of possible values for url.

```

## Examples

### Successfully deleting a resource

The following request asks the server to delete the resource `file.html`:

```http
DELETE /file.html HTTP/1.1
Host: example.com
```

If the request is successful, there are several possible [successful response status codes](/en-US/docs/Web/HTTP/Status#successful_responses).

> A {{HTTPStatus("204", "204 No Content")}} means the request has been enacted by the server and no additional information needs to be sent back to the client:

Reviewer: Or 

```suggestion
A {{HTTPStatus("204", "204 No Content")}} means the request was successful and no additional information needs to be sent back to the client:
```

> A {{HTTPStatus("204", "204 No Content")}} means the request has been enacted by the server and no additional information needs to be sent back to the client:

Author: I like the second one here 👍🏻

```http
HTTP/1.1 204 No Content
Date: Wed, 04 Sep 2024 10:16:04 GMT
```

> A {{HTTPStatus("200", "200 OK")}} means the request has been enacted by the server and the response body includes a representation describing the outcome:

Reviewer: Ditto, maybe

```suggestion
A {{HTTPStatus("200", "200 OK")}} means the request was successful and the response body includes a representation describing the outcome:
```

```http
HTTP/1.1 200 OK
Content-Type: text/html; charset=UTF-8
Date: Fri, 21 Jun 2024 14:18:33 GMT
Content-Length: 1234

<html>
  <body>
    <h1>File "file.html" deleted.</h1>
  </body>
</html>
```

> A {{HTTPStatus("202", "202 Accepted")}} means the request will likely succeed but has not yet been enacted by the server:

Reviewer: Ah, so this is why enacted was used. It still sux. Maybe:

```suggestion
A {{HTTPStatus("202", "202 Accepted")}} means the request has been accepted and will probably succeed but the resource has not yet been deleted by the server.
```
